### PR TITLE
Feature - add overriding config file

### PR DESCRIPTION
**Pull request by daveteu**
**Description of the change**

Feature request: https://github.com/bitnami/bitnami-docker-redis/issues/226

This will check for existence of a mounted `/opt/bitnami/redis/mounted-etc/overrides.conf` file, and append a `include` directives at the end of `redis.conf`.

Nothing will be done if 
- a custom `redis.conf` is  provided.
- user did not mount `/opt/bitnami/redis/mounted-etc/overrides.conf`

### CODE REVIEW

**Comment by daveteu** [comment]
I tried many times but i do not know how to fix the fail check. Hope it doesn't affect the pull request.

**Comment by carrodher** [comment]
Thanks for creating this PR. Please, note it is needed to sign the commits. Your signature certifies that you wrote the patch or otherwise have the right to contribute the material. You can find more info about this topic, in [the “Sign your work” section from the contributing guidelines](https://github.com/bitnami/bitnami-docker-ASSET/blob/master/CONTRIBUTING.md#sign-your-work).

**Review by marcosbc — CHANGES_REQUESTED** [review]
Thanks for the PR! I think it could make sense to allow some extra configuration to the Redis configuration file.

However, wouldn't it make more sense to have a variable like `REDIS_EXTRA_CONF`? That way, if specified, the contents stored in the environment variable would be appended directly to `redis.conf`.

For example, if you wanted to include a file (although it could be easier just to store the expected file contents inside the envvar), you would do this:

```
REDIS_EXTRA_CONF="include /bitnami/redis-mounted/redis.conf"
```

And if you add the envvar to the `redis_env_vars` list in `redis-env.sh`, you can simply specify the following (the code will automatically define `REDIS_EXTRA_CONF` with the contents of the file specified in `REDIS_EXTRA_CONF_FILE`):

```
REDIS_EXTRA_CONF_FILE="/bitnami/redis-mounted/redis.conf"
```

And with that, the contents of that file would automatically be appended to `redis.conf`, only by defining the `REDIS_EXTRA_CONF` variable.

Does it make sense to you?

**Comment by daveteu** [comment]
I think you meant exposing `REDIS_EXTRA_CONF`. The reason why I did not went with `REDIS_EXTRA_CONF="include ...."` route is because I don't think we need users to declare yet another environment variable.

What I proposed is similiar to custom `redis.conf`, if you have a custom redis.conf, you just bind it to /opt/bitnami/redis/mounted-etc/redis.conf.

Likewise, if you have overrides,  just bind to /opt/bitnami/redis/mounted-etc/overrides.conf, there isn't a need to declare additional env variables.

Open to discussion.

**Comment by marcosbc** [comment]
Not sure what you meant about this (since my understanding is that adding the one you proposed would have the same effect):

> I think you meant exposing `REDIS_EXTRA_CONF`? The reason why I did not went with `REDIS_EXTRA_CONF="include ...."` route is because I don't think we need users to declare yet another environment variable.

As for separating with commas, I don't think that is going to be necessary, it would be enough to support newlines (since environment variables do support newlines).

That said, if the concern is not adding an additional environment variable, we could always define a path inside `/bitnami/redis/mounted-etc` that would allow for mounting extra configurations to Redis. So, if the file exists, an "include" could be added to the config file. WDYT?

**Comment by daveteu** [comment]
> That said, if the concern is not adding an additional environment variable, we could always define a path inside `/bitnami/redis/mounted-etc` that would allow for mounting extra configurations to Redis. So, if the file exists, an "include" could be added to the config file. WDYT?

This is what this pull request is for. The script will check for existence of `/opt/bitnami/redis/mounted-etc/overrides.conf`, if it exists, add an `include opt/bitnami/redis/mounted-etc/overrides.conf` line at the very end of `redis.conf`.

**Review by marcosbc — COMMENTED** [review]
I see, thank you for clarifying that! If so, this PR makes sense. I was a bit confused with making `REDIS_OVERRIDES_FILE` a variable, but given that it is indeed acting as a constant value, it makes total sense.

I added a minor request below, the rest LGTM. Once you fix it, we can proceed with the merge and release of these changes.

File: 5.0/debian-10/rootfs/opt/bitnami/scripts/redis-env.sh [hunk 1]
```diff
@@ -97,6 +98,7 @@ export ALLOW_EMPTY_PASSWORD="${ALLOW_EMPTY_PASSWORD:-no}"
 export REDIS_PASSWORD="${REDIS_PASSWORD:-}"
 export REDIS_MASTER_PASSWORD="${REDIS_MASTER_PASSWORD:-}"
 export REDIS_ACLFILE="${REDIS_ACLFILE:-}"
+export REDIS_OVERRIDES_FILE="${REDIS_OVERRIDES_FILE:-}"
```
[the remark this answers is not in the document]
Comment (daveteu) [reply 5.0/debian-10/rootfs/opt/bitnami/scripts/redis-env.sh]: Change made, kindly review.

**Review by marcosbc — APPROVED** [review]
LGTM! Thanks for the great contribution!

**Pull request merged**